### ISSUE TITLE
Named Arguments

### ISSUE THREAD

**Issue by kubawerlos** [thecodingmachine/safe#324]
Is breaking straightforward migration when [named arguments](https://wiki.php.net/rfc/named_params) are used a bug or feature?

E.g. this code - without Safe - will work fine:
```php
    \json_decode($json, flags: \JSON_THROW_ON_ERROR);
```
migration to using Safe would require to changing `flags` to `options` though.

**Comment by dbrekelmans** [comment]
Which PHP version and which safe version are you using?

**Comment by kubawerlos** [comment]
Both latest:
```php
echo PHP_VERSION_ID, PHP_EOL,
    json_decode(file_get_contents(__DIR__. '/composer.lock'))->packages[0]->name, PHP_EOL,
    json_decode(file_get_contents(__DIR__. '/composer.lock'))->packages[0]->version, PHP_EOL,
    var_export((new \ReflectionFunction('Safe\json_decode'))->getParameters(), true), PHP_EOL,
    var_export((new \ReflectionFunction('\json_decode'))->getParameters(), true);
```

```console
80102
thecodingmachine/safe
v2.1.1
array (
  0 => 
  ReflectionParameter::__set_state(array(
     'name' => 'json',
  )),
  1 => 
  ReflectionParameter::__set_state(array(
     'name' => 'assoc',
  )),
  2 => 
  ReflectionParameter::__set_state(array(
     'name' => 'depth',
  )),
  3 => 
  ReflectionParameter::__set_state(array(
     'name' => 'options',
  )),
)
array (
  0 => 
  ReflectionParameter::__set_state(array(
     'name' => 'json',
  )),
  1 => 
  ReflectionParameter::__set_state(array(
     'name' => 'associative',
  )),
  2 => 
  ReflectionParameter::__set_state(array(
     'name' => 'depth',
  )),
  3 => 
  ReflectionParameter::__set_state(array(
     'name' => 'flags',
  )),
)
```

**Comment by dbrekelmans** [comment]
Ah this is a bug then. Since `json_decode` is a special case function, it's not automatically updated when the phpdocs are changed.

If you are up for it, you can submit a PR changing the parameter names in https://github.com/thecodingmachine/safe/blob/master/lib/special_cases.php

**Comment by MarcinGladkowski** [comment]
@dbrekelmans Hi! Consider my changes, please. https://github.com/thecodingmachine/safe/pull/376

**Comment by dbrekelmans** [comment]
Fixed in #376
